app/models/user.rb

```diff
@@ -123,33 +118,33 @@ def create_profile
   has_many :articles, dependent: :destroy
   has_many :audit_logs, dependent: :nullify
   has_many :authored_notes, inverse_of: :author, class_name: "Note", foreign_key: :author_id, dependent: :delete_all
-  has_many :backup_data, foreign_key: "instance_user_id",
-                         inverse_of: :instance_user, class_name: "BackupData", dependent: :delete_all
+  has_many :backup_data, foreign_key: "instance_user_id", inverse_of: :instance_user,
+                         class_name: "BackupData", dependent: :delete_all
   has_many :badge_achievements, dependent: :destroy
   has_many :badges, through: :badge_achievements
-  has_many :blocked_blocks, class_name: "UserBlock",
-                            foreign_key: :blocked_id, inverse_of: :blocked, dependent: :delete_all
-  has_many :blocker_blocks, class_name: "UserBlock",
-                            foreign_key: :blocker_id, inverse_of: :blocker, dependent: :delete_all
+  has_many :blocked_blocks, class_name: "UserBlock", foreign_key: :blocked_id,
+                            inverse_of: :blocked, dependent: :delete_all
+  has_many :blocker_blocks, class_name: "UserBlock", foreign_key: :blocker_id,
+                            inverse_of: :blocker, dependent: :delete_all
   has_many :chat_channel_memberships, dependent: :destroy
   has_many :chat_channels, through: :chat_channel_memberships
-  has_many :listings, dependent: :destroy
-  has_many :endorsements, dependent: :destroy, class_name: "ListingEndorsement"
   has_many :collections, dependent: :destroy
   has_many :comments, dependent: :destroy
   has_many :created_podcasts, class_name: "Podcast", foreign_key: :creator_id, inverse_of: :creator, dependent: :nullify
   has_many :credits, dependent: :destroy
   has_many :display_ad_events, dependent: :destroy
   has_many :email_authorizations, dependent: :delete_all
   has_many :email_messages, class_name: "Ahoy::Message", dependent: :destroy
+  has_many :endorsements, dependent: :destroy, class_name: "ListingEndorsement"
   has_many :field_test_memberships, class_name: "FieldTest::Membership", as: :participant, dependent: :destroy
   has_many :github_repos, dependent: :destroy
   has_many :html_variants, dependent: :destroy
   has_many :identities, dependent: :destroy
   has_many :identities_enabled, -> { enabled }, class_name: "Identity", inverse_of: false
+  has_many :listings, dependent: :destroy
   has_many :mentions, dependent: :destroy
   has_many :messages, dependent: :destroy
-  has_many :notes, as: :noteable, inverse_of: :noteable
+  has_many :notes, as: :noteable, inverse_of: :noteable, dependent: :destroy
```

Comment:
this is what's new, `dependent: :destroy`. The rest is reordering alphabetically